### ISSUE TITLE
Type narrowing does not work after simple inclusion check

### ISSUE THREAD

**Issue by lovetox** [microsoft/pyright#2666]
**Describe the bug**
pyright is not able to narrow type after simple inclusion check

GeneralTypeIssue is raised

**Expected behavior**
name should be narrowed to string if the condition is true

**Screenshots or Code**
```python
from typing import Optional

def test(name: Optional[str]) -> str:
    if name in ('a', 'b'):
        return name
    return 'c'

```

**VS Code extension or command-line**
Sublime LSP-pyright

**Additional context**
pyright 1.1.192
mypy --strict passes without errors

**Comment by erictraut** [comment]
This isn't currently a supported type narrowing pattern. For a full list of supported type narrowing patterns, refer to [this documentation](https://github.com/microsoft/pyright/blob/main/docs/type-concepts.md#type-guards).

Each new type narrowing pattern requires specialized code that adds complexity and analysis overhead. We do occasionally add support for new patterns, but we do so only when we have strong signal that it's a sufficiently common pattern. We use two sources for this signal:
1. Upvotes on feature requests (in the form of thumbs up emoticons)
2. Support for the same pattern in other static type checkers like mypy

Mypy appears to support this particular case, so that meets the second criteria.

**Comment by erictraut** [comment]
This will be included in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.194, which I just published. It will also be included in the next release of pylance.
